File: Easy/Leaders in an array/leaders-in-an-array.cpp
//{ Driver Code Starts
// C++ program to remove recurring digits from
// a given number
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution{
public:
    vector<int> leaders(int a[], int n){
        vector<int> result;
        if(n == 0)
            return result;
        
        int max_right = a[n - 1]; // Rightmost element is always a leader
        result.push_back(max_right);
        
        for(int i = n - 2; i >= 0; i--){
            if(a[i] >= max_right){
                max_right = a[i];
                result.push_back(max_right);
            }
        }
        
        reverse(result.begin(), result.end()); // Reverse the result to maintain the original order
        return result;
    }
};

//{ Driver Code Starts.

int main()
{
   long long t;
   cin >> t;//testcases
   while (t--)
   {
       long long n;
       cin >> n;//total size of array
        
        int a[n];
        
        //inserting elements in the array
        for(long long i =0;i<n;i++){
            cin >> a[i];
        }
        Solution obj;
        //calling leaders() function
        vector<int> v = obj.leaders(a, n);
        
        //printing elements of the vector
        for(auto it = v.begin();it!=v.end();it++){
            cout << *it << " ";
        }
        
        cout << endl;

   }
}

// } Driver Code Ends

File: Easy/Leaders in an array/leaders-in-an-array.py

class Solution:
    #Function to find the leaders in the array.
    def leaders(self, A, N):
        leaders_list = []
        max_right = float('-inf')  # Initialize max_right as negative infinity
        
        # Traverse the array from right to left
        for i in range(N - 1, -1, -1):
            if A[i] >= max_right:  # Check if the current element is greater than or equal to max_right
                leaders_list.append(A[i])  # If yes, it's a leader
                max_right = A[i]  # Update max_right
        
        # Reverse the leaders_list to maintain the order of appearance
        return leaders_list[::-1]

#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math


    
def main():
    
    T=int(input())
    
    while(T>0):
        
        
        N=int(input())
        
        A=[int(x) for x in input().strip().split()]
        obj = Solution()
        
        A=obj.leaders(A,N)
        
        for i in A:
            print(i,end=" ")
        print()
        
        T-=1

if __name__=="__main__":
    main()
# } Driver Code Ends